ui/plugin-system/src/runtime/annotations.ts

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { AnnotationEvent } from '@perses-dev/core';
+import { createContext, useContext } from 'react';
+
+export type AnnotationSrv = {
+  annotations: AnnotationEvent[];
+};
+
+export const AnnotationContext = createContext<AnnotationSrv | undefined>(undefined);
+
+export function useAnnotationContext() {
+  const ctx = useContext(AnnotationContext);
+  if (ctx === undefined) {
+    throw new Error('No found AnnotationContext. Did you forget the AnnotationsProvider?');
```

Comment:
I know we've done this with most of the others providers so far, but does AnnotationsProvider need need to be required or is there some way to fallback gracefully without the error?

Reply:
Yeah I guess I go back and forth on this. If we end up creating one provider that has all the presets I think this should be ok. If not, I think we should make it optional.

`<PersesProvider />` should setup the core providers